### ISSUE TITLE
Failing Unit Tests after rebase

### ISSUE THREAD

**Issue by Dinistro** [Superty/llvm-project#16]
As I already informed @Superty, the following test are failing on `feature-expr-from-set` :

```
MLIR-Unit :: Analysis/./MLIRAnalysisTests/FlatAffineConstraintsTest.FindSampleTest
MLIR-Unit :: Analysis/./MLIRAnalysisTests/FlatAffineConstraintsTest.IsIntegerEmptyTest
```

Both of them are violating the same assertion:

```
mlir/lib/Analysis/Presburger/Simplex.cpp:929: 
mlir::Fraction mlir::GBRSimplex::computeWidthAndDuals(ArrayRef<int64_t>, SmallVectorImpl<int64_t> &, int64_t &): 
Assertion `maybeWidth.hasValue() && "Width should not be unbounded!"' failed.
```

The problem started appearing after the commit "feature: Simplex::killCol and Simplex::closeRow": https://github.com/Superty/llvm-project/commit/8110c33bf22b19cbc7ae5b1e506e82d25aa253c6

This commit introduces not only `killCol` and `closeRow` but also changes in `makeProduct` that I wasn't able to understand. Actually I'm even somewhat sure that this is the place where we / I broke it. 
Note that `makeProduct` is executed in the constructor of `GBRSimplex` and there is not much more functionality that could break with this commit.

**Comment by Dinistro** [comment]
Another cause of the problem could be that `GBRSimplex` has not a single change in the linked commit. I assume that dead columns should be treated differently than alive columns.

**Comment by superty** [comment]
I think this is most likely because of `makeProduct` not handling zero'd columns or redundant rows properly. This is just a heuristic optimisation, which ideally I plan to get rid of to make our patch as small as possible for upstreaming. I will look into doing so next week. Hopefully that will fix the issue.

**Comment by Dinistro** [comment]
I thought the same, but the failing tests do not have redundant rows nor zero'd columns it seems.

**Comment by superty** [comment]
I still think the issue is probably in the changes made in order to handle these. Reverting these changes and removing the heuristics that make them necessary would be my next step. If it doesn't work, then the next option would be debugging the code in detail, I guess.

**Comment by superty** [comment]
Should be fixed by b7b48e7a5e2d4fd380fb18058718c6c6f5116815.

**Comment by Dinistro** [comment]
Thank you very much. I will move the commit the fixes the other issue on to the right branch as well.
